android/src/com/frostwire/android/gui/SearchEngine.java

```diff
@@ -249,10 +249,33 @@ public SearchPerformer getPerformer(long token, String keywords) {
     };
 
     public static final SearchEngine ONE337X = new SearchEngine("1337x", Constants.PREF_KEY_SEARCH_USE_ONE337X) {
+        private String domainName = null;
+
         @Override
         public SearchPerformer getPerformer(long token, String keywords) {
-            return new One337xSearchPerformer("www.1377x.to", token, keywords, DEFAULT_TIMEOUT);
-            // change www.1377x.to to other like www.1337x.gd, www.1337x.to or other proxy site etc
+            if (domainName == null) {
+                throw new RuntimeException("check your logic, this search performer has no domain name ready");
+            }
+            return new One337xSearchPerformer(domainName, token, keywords, DEFAULT_TIMEOUT);
+        }
+
+        protected void postInitWork() {
+            new Thread(() -> {
```

Comment:
Seeing this again gives me an idea (not for the scope of this patch)

Now we have 2 extra threads on startup to determine these mirrors.

Perhaps we should use an existing or new executor that deals with mirror picking so that we don't have to create expensive new threads, we can recycle unused threads from an existing executor/pool.

These `postInitWork()` methods logic could be rethought to work instead in a lazy manner, the performer could subscribe to mirror picking event dispatcher, the event dispatcher sends an event when it's ready with the fastest mirror domain.

(just thinking out loud)

I'm thinking, in the `postInitWork()` in the future we'll do something like this (using a lambda for the listener):

```java
FastMirrorPicker.getFastestMirrorDomain(
  mirrors, 
  (fastestMirrorResult) -> ONE337X.domainName = fastestMirrorResult);
```

The `FastMirrorPicker` class deals with all the details of dealing with the thread pool, using the right httpClient instance, setting proper timeouts, etc.

Reply:
I actually have no experience with Threads and lambda function 😅😅.
just a workaround : 
SearchEngine.java <img src="https://user-images.githubusercontent.com/32963903/86487801-ba744d80-bd7c-11ea-9690-cc11a028d868.png" width="720" height="20">
<img src="https://user-images.githubusercontent.com/32963903/86487763-9b75bb80-bd7c-11ea-8521-2d20daf1994b.png"  width="720" height="480">
URLUTILS.java
<img src="https://user-images.githubusercontent.com/32963903/86488584-18099980-bd7f-11ea-8df2-f38f8e8a3886.png" width="720" height="440">

but i think it will break the desktop code ☠

Reply:
"and implementing the postInitWork() function before SearchEngine is initiated"

This is an interesting idea, but then the search engine object couldn't be added to the global list of search engines, or such list could not be created until we have those search engines with mirrors ready...